docs/reviewer-guide.md

```diff
@@ -41,6 +41,58 @@ Make a few passes over the code you want to review.
 * List ordering on CRD APIs matter. If two components need to update a list on the same object, make sure both components do it in a way that preserves the order of the list. For example, both virt-handler and virt-controller need to modify conditions on the VMI status. If both virt-handler and virt-controller are constantly changing the order of the conditions list, that will cause an update storm where both components are competing with one another to write changes.
 * Privileged node-level operations should be added to virt-handler and not virt-launcher to keep the privileges on virt-launcher at a minimum.
 
+## In-Depth PATCH/UPDATE considerations
+
+When trying to determine if an UPDATE or a PATCH should be used, the following has to be considered:
+
+### Updating `spec` or `metadata` sections with a controller
+
+Did the controller in question CREATE the object it operates on?
+- Yes: `Update` can be used
```

Comment:
So, that `Update` can still fail, if for example another controller patched the object between our `Get` and our `Update`, right?
If that's accurate, shouldn't we still _recommend_ using `Patch` (and `UpdateStatus`) even on objects we created?

Reply:
I think it is an important property of the system that you get a conflict notification if the object updated in the meantime. With patch you can easily end up in the situation where you patch the same object multiple times because you did not see yet in your watcher that the object got transformed already. We can make use of `expectations` but they are a more advanced topic.